config/default/kustomization.yaml

```diff
@@ -13,6 +13,7 @@ namePrefix: baremetal-operator-
 #  someName: someValue
 
 bases:
+- ../namespace
```

Comment:
I wonder https://github.com/metal3-io/baremetal-operator/blob/24103e39f5596873ee5820663d05487fadeee81d/config/default/kustomization.yaml#L9 prefix will also apply namespace. As far as I recall, this is the reason namespace is not used in here.

Reply:
Hi @ardaguclu . Sorry for my late reply, I was on vacation for some time.
Actually we don't need namespace manifest below to be added in config/render/capm3.yaml, because we already have it in https://github.com/metal3-io/baremetal-operator/blob/main/config/namespace/namespace.yaml and here we are just reusing it. This is the same as in case of [basic-auth](https://github.com/metal3-io/baremetal-operator/blob/23493e6d94bbcff1c4da4623197674022dd5ccae/config/basic-auth/default/kustomization.yaml#L6) or [tls](https://github.com/metal3-io/baremetal-operator/blob/23493e6d94bbcff1c4da4623197674022dd5ccae/config/tls/kustomization.yaml#L6). So, it was required to install the namespace only when using config/default (non tls or basic-auth). 
About the label, we already have the label for all the resource like [Deployment](https://github.com/metal3-io/baremetal-operator/blob/23493e6d94bbcff1c4da4623197674022dd5ccae/config/manager/manager.yaml#L7) and other objects related to the BMO deployment including the [namespace](https://github.com/metal3-io/baremetal-operator/blob/23493e6d94bbcff1c4da4623197674022dd5ccae/config/manager/manager.yaml#L7) that we are trying to reuse here. And that's perhaps for some sort of filtering, I'm not sure TBH.